tests/mod.rs

```diff
@@ -618,3 +634,13 @@ fn bit_vec_works() {
 	let v2 = MyStruct::decode_all(&mut &v2[..]).unwrap();
 	assert_eq!(v1.x, v2.x);
 }
+
+#[test]
+fn no_warning_for_deprecated() {
+	#[derive(Encode, Decode)]
+	pub enum MyEnum {
+		VariantA,
+		#[deprecated]
+		VariantB,
+	}
+}
```

Comment:
Nit: even if the warning is there this will still compile, right? It'd be nice to maybe have a compile test that'd enforce that the warning's not there, but it's not really a big deal.